discord-bot/src/helpers.ts

```diff
@@ -0,0 +1,235 @@
+}
+
+export const handleBet = async (
+  reaction: MessageReaction,
+  user: User,
+  channel: TextChannel,
+  message: Message,
+  market: FullMarket,
+  sale?: boolean
+) => {
+  const emojiKey = getBetEmojiKey(reaction)
+  if (!emojiKey) return
+  const { amount, outcome: buyOutcome } = bettingEmojis[emojiKey]
+  console.log('betting', amount, buyOutcome, 'on', market.id, 'for', user.tag)
+  try {
+    const apiKey = await userApiKey(user.id)
+    if (!apiKey) {
+      if (sale) return
+      await user.send(registerHelpMessage)
+      const userReactions = message.reactions.cache.filter((reaction) =>
+        reaction.users.cache.has(user.id)
+      )
+      try {
+        for (const reaction of userReactions.values()) {
+          await reaction.users.remove(user.id)
+        }
+      } catch (error) {
+        console.error('Failed to remove reactions.')
+      }
+      return
+    }
+
+    const outcome = sale ? (buyOutcome === 'YES' ? 'NO' : 'YES') : buyOutcome
+    // send json post request to api
+    const resp = await fetch('https://manifold.markets/api/v0/bet', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Key ${apiKey}`,
+      },
+      body: JSON.stringify({
+        amount,
+        contractId: market.id,
+        outcome,
+      }),
+    })
+    if (!resp.ok) {
+      const content = `Error: ${resp.statusText}`
+      await sendThreadMessage(channel, market, content, user)
+      return
+    }
+    const bet = await resp.json()
+    const newProb = bet.probAfter
+    const content = `${user.tag} ${
+      sale ? 'sold' : 'bet'
+    } M$${amount} on ${buyOutcome}. New probability: ${Math.round(
+      newProb * 100
+    )}%`
+    market.probability = newProb
+    await sendThreadMessage(channel, market, content, user)
+    await updateMarketStatus(message, market)
+  } catch (e) {
+    const content = `Error: ${e}`
+    await sendThreadMessage(channel, market, content, user)
+  }
+}
+const currentProbText = (prob: number) =>
+  `**${Math.round(prob * 100)}%** chance`
+
+export const getCurrentMarketDescription = (market: FullMarket) => {
+  const closed = (market.closeTime ?? 0) <= Date.now()
+  let content = currentProbText(market.probability ?? 0)
+  if (closed) {
+    content = market.isResolved
+      ? `Market resolved ${market.resolution}`
+      : `Market closed at ${Math.round(
+          (market.probability ?? 0) * 100
+        )}% chance.`
+  }
+  return content
+}
+
+const updateMarketStatus = async (message: Message, market: FullMarket) => {
+  const previousEmbed = message.embeds[0]
+  const marketEmbed = EmbedBuilder.from(previousEmbed)
+  marketEmbed.setDescription(getCurrentMarketDescription(market))
+  await message.edit({ embeds: [marketEmbed], files: [] })
+}
+
+const getThread = async (
+  channel: TextChannel,
+  marketName: string,
+  title?: string
+) => {
+  const name = marketName.slice(0, 40) + '...'
```

Comment:
This seems like it's going to run into trouble, on markets that are e.g. like "Will blah blah blah by 2023", "Will blah blah blah by 2024", etc.

You could store a market message ID -> thread message ID mapping in Supabase.

Reply:
Good idea! Done.